test_conformance/vulkan/main.cpp

```diff
@@ -0,0 +1,275 @@
+    if(argCount == 0 )
+        return 0;
+    // get the platform ID
+    errNum = clGetPlatformIDs(1, &platform, NULL);
+    if (errNum != CL_SUCCESS) {
+        print_error(errNum, "Error: Failed to get platform\n");
+        return errNum;
+    }
+    
+    errNum = clGetDeviceIDs(platform, CL_DEVICE_TYPE_GPU, 0, NULL, &num_devices);
+    if (CL_SUCCESS != errNum) {
+       print_error(errNum, "clGetDeviceIDs failed in returning of devices\n");
+       return errNum;
+    }
+    devices = (cl_device_id *)malloc(num_devices * sizeof(cl_device_id));
+    if (NULL == devices) {
+        print_error(errNum, "Unable to allocate memory for devices\n");
+        return CL_OUT_OF_HOST_MEMORY;
+    }
+    errNum = clGetDeviceIDs(platform, CL_DEVICE_TYPE_GPU, num_devices, devices, NULL);
+    if (CL_SUCCESS != errNum) {
+       print_error(errNum, "Failed to get deviceID.\n");
+       return errNum;
+    }
+    for (device_no = 0; device_no < num_devices; device_no++) {
+        errNum = clGetDeviceInfo(devices[device_no], CL_DEVICE_EXTENSIONS, 0, NULL, &extensionSize);
+        if (CL_SUCCESS != errNum) {
+            log_error("Error in clGetDeviceInfo for getting device_extension size....\n");
+            return errNum;
+        }
+        extensions = (char*)malloc(extensionSize);
+        if (NULL == extensions) {
+            log_error("Unable to allocate memory for extensions\n");
+            return CL_OUT_OF_HOST_MEMORY;
+        }
+        errNum = clGetDeviceInfo(devices[device_no], CL_DEVICE_EXTENSIONS, extensionSize, extensions, NULL/*&extensionSize*/);
+        if (CL_SUCCESS != errNum) {
+            print_error(errNum, "Error in clGetDeviceInfo for getting device_extension \n");
+            return errNum;
+        }
+        errNum = clGetDeviceInfo(devices[device_no], CL_DEVICE_UUID_KHR, CL_UUID_SIZE_KHR, uuid, &extensionSize);
+        if (CL_SUCCESS != errNum) {
+             print_error(errNum, "clGetDeviceInfo failed with error\n ");
+             return errNum;
+        }
+        errNum = memcmp(uuid, vkDevice.getPhysicalDevice().getUUID(), VK_UUID_SIZE);
+        if (errNum == 0) {
+            break;
+        }
+    }
+    if (device_no >= num_devices) {
+         fprintf(stderr, "OpenCL error:"
+               " No Vulkan-OpenCL Interop capable GPU found.\n");
+    }
+    if (!( is_extension_available(devices[device_no], "cl_khr_external_memory")  && is_extension_available(devices[device_no], "cl_khr_external_semaphore"))) {
+        log_info(" Device does not support cl_khr_external_memory or cl_khr_external_semaphore\n");
+        log_info(" TEST SKIPPED\n");
+        return CL_SUCCESS;
+    }
+    init_cl_vk_ext(platform);
+
+ // Execute tests.
+      
+ // Note: don't use the entire harness, because we have a different way of obtaining the device (via the context)
+    errNum = parseAndCallCommandLineTests( argCount, argList, devices[device_no], test_num, test_list,  true, 0, 1024 );
```

Comment:
Maybe we should add support to the harness instead of duplicating the code here?

Reply:
Will consider this in a separate PR.